src/StoreApi/Utilities/ProductQueryFilters.php

```diff
@@ -111,63 +113,161 @@ private function generate_stock_status_count_query( $status, $product_query_sql,
 	}
 
 	/**
-	 * Get attribute counts for the current products.
+	 * Get terms list for a given taxonomy.
 	 *
-	 * @param \WP_REST_Request $request The request object.
-	 * @param array            $attributes Attributes to count, either names or ids.
-	 * @return array termId=>count pairs.
+	 * @param string $taxonomy Taxonomy name.
+	 *
+	 * @return array
 	 */
-	public function get_attribute_counts( $request, $attributes = [] ) {
+	public function get_terms_list( string $taxonomy ) {
```

Comment:
I think it would still cause an issue because you have changed the function signature. I was thinking just use the same practice Woo core does when deprecating. Like this https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-emails.php#L308-L319

Reply:
> I think it would still cause an issue because you have changed the function signature.

Good point, the second param was modified from an `$attributes` array to a `$filtered_attribute` string. How about keeping things contained and deprecating [the modified second param](https://github.com/woocommerce/woocommerce-blocks/pull/8599/commits/640f66fd4828b559afe36694d0bc092fe3250b7e)?